src/codegen/zod-utils.ts

```diff
@@ -433,6 +433,10 @@ export const ZodUtils = {
       return `z.object({${props}})`
     }
 
+    if (def.typeName === 'ZodBranded') {
+      return def.description
```

Comment:
Yes, this means the description is load-bearing. I don't think this overly matters since this is an internal-only description BUT we could certainly do property detection here (e.g. return `PrefabDurationSchema` if `duration` in properties).